### PR TITLE
Implement the interface required by the foreign-toplevel protocol for xwayland

### PR DESCRIPTION
**Pull request by wineee**
Treeland impl: https://github.com/linuxdeepin/treeland/pull/196

### CODE REVIEW

File: src/server/kernel/wtoplevelsurface.h [hunk 1]
```diff
@@ -49,6 +51,12 @@ class WToplevelSurface : public QObject
     virtual bool isFullScreen() const {
         return false;
     }
+    virtual QString title() const {
+        return "";
```
Comment (zccrs) [inline src/server/kernel/wtoplevelsurface.h:55]: return {};

File: src/server/kernel/wtoplevelsurface.h [hunk 2]
```diff
@@ -49,6 +51,12 @@ class WToplevelSurface : public QObject
...
+    };
+    virtual QString appId() const {
+        return "";
```
Comment (zccrs) [inline src/server/kernel/wtoplevelsurface.h:58]: return {}

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/kernel/wxwaylandsurface.cpp [hunk 3]
```diff
@@ -392,6 +395,27 @@ QRect WXWaylandSurface::getContentGeometry() const
     return QRect(0, 0, d->nativeHandle()->width, d->nativeHandle()->height);
 }
 
+QString WXWaylandSurface::title() const
+{
+    W_DC(WXWaylandSurface);
+
+    return { d->nativeHandle()->title };
```
Comment (zccrs) [inline src/server/kernel/wxwaylandsurface.cpp:402]: QString::fromUtf8 ?

What's the text encoding of title?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/server/kernel/wxwaylandsurface.cpp (hunk 3 above)
Comment (wineee) [reply src/server/kernel/wxwaylandsurface.cpp:402]: copy from xdgsurface @justforlxz

Comment (wineee) [reply src/server/kernel/wxwaylandsurface.cpp]: It must be  utf8

**Review by zccrs — APPROVED** [review]

**Pull request merged**